haskell/haskell.bzl

```diff
@@ -130,13 +130,21 @@ def _mk_binary_rule(**kwargs):
         ),
     )
 
-    # Tests have an extra fields regarding expected code coverage percentages.
+    # Tests have an extra fields regarding code coverage.
     if is_test:
         attrs.update({
-            "expected_expression_coverage": attr.int(
-                default = 0,
+            "expected_covered_expressions_percentage": attr.int(
+                default = -1,
                 doc = "The expected percentage of expressions covered by testing.",
             ),
+            "expected_uncovered_expression_count": attr.int(
+                default = -1,
+                doc = "The expected number of expressions which are not covered by testing.",
+            ),
+            "strict_coverage_analysis": attr.bool(
```

Comment:
Shouldn't this be `True` by default?

Reply:
No, I don't think the behaviour described by strict code coverage is what people expect by default. People expect to be setting minimums/maximums, not exact values. There is value, however, in setting exact values if your team is willing to constantly re-set them for every change (namely, it forces awareness and constant improvement of test coverage).